drivers/web-worker-driver/src/wasmJsMain/kotlin/app/cash/sqldelight/driver/worker/WebWorkerDriver.kt

```diff
@@ -0,0 +1,240 @@
+package app.cash.sqldelight.driver.worker
+
+import app.cash.sqldelight.Query
+import app.cash.sqldelight.Transacter
+import app.cash.sqldelight.db.QueryResult
+import app.cash.sqldelight.db.SqlCursor
+import app.cash.sqldelight.db.SqlDriver
+import app.cash.sqldelight.db.SqlPreparedStatement
+import app.cash.sqldelight.driver.worker.api.WorkerAction
+import app.cash.sqldelight.driver.worker.api.WorkerActions
+import app.cash.sqldelight.driver.worker.api.WorkerRequest
+import app.cash.sqldelight.driver.worker.api.WorkerResponse
+import app.cash.sqldelight.driver.worker.api.WorkerResult
+import app.cash.sqldelight.driver.worker.util.add
+import app.cash.sqldelight.driver.worker.util.instantiateObject
+import app.cash.sqldelight.driver.worker.util.isArray
+import app.cash.sqldelight.driver.worker.util.jsonStringify
+import app.cash.sqldelight.driver.worker.util.objectEntries
+import app.cash.sqldelight.driver.worker.util.toJsArray
+import app.cash.sqldelight.driver.worker.util.toUint8Array
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.khronos.webgl.Uint8Array
+import org.khronos.webgl.get
+import org.w3c.dom.MessageEvent
+import org.w3c.dom.Worker
+import org.w3c.dom.events.Event
+
+/**
+ * A [SqlDriver] implementation for interacting with SQL databases running in a Web Worker.
+ *
+ * This driver is dialect-agnostic and is instead dependent on the Worker script's implementation
+ * to handle queries and send results back from the Worker.
+ *
+ * @property worker The Worker running a SQL implementation that this driver communicates with.
+ * @see [WebWorkerDriver.fromScriptUrl]
+ */
+class WebWorkerDriver(private val worker: Worker) : SqlDriver {
```

Comment:
Which elements of this class (and its `jsMain` counterpart) can't be lifted into the `commonMain` implementation? Ideally I think we would have one common implementation of the worker driver, and then `expect`/`actual` the web API bits (i.e. `Worker`) as needed.

Reply:
TBH, most of it =)
I have tried to make it as common as possible, but found out that currently there's no way to do it properly.

Most of JS APIs are not common between `js` and `wasmJs`.
Current `wasmJs` interop is quite limited.
It does not support: `Array, dynamic` in external types.
This means that we cannot share `WorkerRequest`, `WorkerResult`, `WorkerResponse`.
Also, we would need to expect/actual quite big portion of `Worker` API with custom wrappers all over the place.

I will try some more ways to commonise as much as possible and show you in a couple of days, maybe we will find a better solution.

Reply:
@dellisd I've tried to commonize `WebWorkerDriver` as much as possible. Please take a look 🙂  https://github.com/cashapp/sqldelight/pull/4965/commits/ece537a35558278801051c2b74c8b9dc91b0d433